repositories/src/main/scala/cromwell/pipeline/datastorage/formatters/package.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.pipeline.datastorage
+
+import play.api.libs.json.{ Format, JsResult, JsValue, Json }
+import cats.Show
+import cromwell.pipeline.model.validator.Wrapped
+import cromwell.pipeline.model.wrapper.{ Name, Password, UserEmail, UserId, VersionValue }
+
+import scala.language.implicitConversions
+import cats.data.Validated.{ Invalid, Valid }
+import cats.implicits._
+import cromwell.pipeline.model.validator.MicrotypeException.MicrotypeValidationException
+import cromwell.pipeline.model.validator.WrappedFactoryPattern
+
+package object formatters {
+  private def wrapperFormat[T, E, W <: Wrapped[T]](wrapperFactory: WrappedFactoryPattern[T, E, W])(
+    implicit show: Show[E],
+    format: Format[T]
+  ): Format[W] =
+    new Format[W] {
+      override def reads(json: JsValue): JsResult[W] = json
+        .validate[T]
+        .map[W](
+          value =>
+            wrapperFactory.from(value) match {
+              case Valid(wrapped) => wrapped
+              case Invalid(e)     => throw MicrotypeValidationException("Microtype is not valid: " + e)
```

Comment:
```suggestion
        .flatMap[W] {
          value =>
            wrapperFactory.from(value) match {
              case Valid(wrapped) => JsSuccess(wrapped)
              case Invalid(e)     => JsError(MicrotypeValidationException(s"Microtype is not valid: $e") }
```

Reply:
ok, thanks, I misunderstood, I thought that you mean drop out this exception at all